pom.xml

```diff
@@ -265,9 +260,33 @@
         <artifactId>error_prone_annotations</artifactId>
         <version>2.10.0</version>
       </dependency>
+      <dependency>
+        <groupId>com.google.inject.extensions</groupId>
+        <artifactId>guice-assistedinject</artifactId>
+        <version>5.0.1</version>
+      </dependency>
     </dependencies>
   </dependencyManagement>
 
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-enforcer-plugin</artifactId>
+        <configuration>
+          <rules combine.children="append">
+            <requireSameVersions>
```

Comment:
Should this be in core if it is tightly coupled to guice version (at least until guice gets removed from core 🦄)

Reply:
> Should this be in core if it is tightly coupled to guice version

It could be, though I suppose core does not use it; or it could be added to the core BOM.

> at least until guice gets removed from core

Or I can remove jclouds from this plugin: https://github.com/jenkinsci/artifact-manager-s3-plugin/pull/215#discussion_r656197429

Reply:
> It could be, though I suppose core does not use it; or it could be added to the core BOM.

if it was just in the BOM then it can constantly drift (core gets a new Guice, this plugin potentially breaks).

> Or I can remove jclouds from this plugin: #215 (comment)

probably better.

Reply:
> core gets a new Guice, this plugin potentially breaks

Potentially, though the test failure I saw was about core bundling Guice 5 and this plugin bundling Guice Assisted Inject 4; I hope breaking changes are limited to major versions which should not be too common. The Enforcer rule will also alert us to problems if are updating the core dep for other reasons. (_Might_ also work in PCT, not sure.)